llama.cpp

```diff
@@ -7461,6 +7447,21 @@ void llama_sample_repetition_penalties(
     }
 }
 
+static std::string llama_token_to_piece(const struct llama_context* ctx, llama_token token) {
+    std::vector<char> result(8, 0);
+    const int n_tokens = llama_token_to_piece(llama_get_model(ctx), token, result.data(), result.size());
+    if (n_tokens < 0) {
+        result.resize(-n_tokens);
+        int check = llama_token_to_piece(llama_get_model(ctx), token, result.data(), result.size());
+        GGML_ASSERT(check == -n_tokens);
+    }
+    else {
+        result.resize(n_tokens);
+    }
+
+    return std::string(result.data(), result.size());
+}
```

Comment:
Any reason to move this down here? Prefer to have it at the top with the rest of the llama helpers

Reply:
Moved it because `grammar` stuff seems to be the only consumer.

Reply:
Yes, and we can merge after